backend/azure/fileSystem.go

```diff
@@ -0,0 +1,138 @@
+package azure
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/c2fo/vfs/v5"
+	"github.com/c2fo/vfs/v5/backend"
+	"github.com/c2fo/vfs/v5/utils"
+)
+
+// Scheme defines the scheme for the azure implementation
+const Scheme = "https"
+
+// Name defines the name for the azure implementation
+const Name = "azure"
+
+const errNilFileSystemReceiver = "azure.FileSystem receiver pointer must be non-nil"
+
+// FileSystem implements the vfs.FileSystem interface for Azure Blob Storage
+type FileSystem struct {
+	options Options
+	client  Client
```

Comment:
Good question.  The client is just a wrapper around the Azure storage sdk and the Azure storage sdk seems to be a wrapper around a microsoft rest library.  Since everything is just a rest call under the hood, there aren't any persistent connections to maintain (or any other shared resource for that matter).  The singleton just holds the common Pipeline object (which doesn't seem to do anything other than hold some configuration) so it doesn't have to be initialized on every call.